src/jesse_database.erl

```diff
@@ -305,11 +305,17 @@ get_schema_info(File, {Acc, ParseFun}) ->
 %% @private
 -spec get_schema_id(Schema :: jesse:json_term()) -> string() | undefined.
 get_schema_id(Schema) ->
-  case jesse_json_path:value(?ID, Schema, undefined) of
+  case jesse_json_path:value(?SCHEMA, Schema, undefined) of
+    ?json_schema_draft6 -> get_schema_id_by_version(?IDv6, Schema);
+                      _ -> get_schema_id_by_version(?ID, Schema)
+  end.
```

Comment:
> but unless I'm mistaken the problem is slightly more complex as Schema doesn't need to be a JSON Schema draft, at least not directly

Yeah, `$schema` meta-schema URIs are technically allowed to be anything, which makes them of questionable use determining syntax and semantics.  An unrecognized meta-schema URI may actually be just like the standard except for a tiny syntax restriction (e.g. forbid the `format` keyword), but there's no programmatic way to know that in draft-06 or draft-07.

Starting in draft 2019-09 (which would have been draft-08, except for [reasons](http://json-schema.org/specification-links.html#understanding-draft-names-and-numbers)), this is handled through the vocabularies concept. `$vocabulary` is a keyword that describes JSON Schema keyword semantics in the instance document.  Which means it is only useful if the instance document is a schema, so in practice, you can have however many meta-schema URIs describing all sorts of combinations of vocabularies and even syntax restrictions (because meta-schemas validate syntax just like before), but there's a small and generally well-known set of vocabulary URIs that tells you what code to actually run.

In the meantime, in practice it's fine to check for draft-04, draft-06, draft-07, etc., or just require the user to specify a draft through configuration or command line options.

Reply:
Right, even more complex than initially thought :) Thanks for your input.
My complexity bounds were regarding the possibility that `$schema` can point to `https://example.com/andreis-schema`, which in turn extends say draft-6.

Reply:
It ends up being fairly simple- `https://example.com/andreis-schema` would list the same standard vocabularies as the standard meta-schema, `allOf` the standard meta-schema, and also describe whatever extension you add.  You can make that extension a vocabulary if you want other implementations to be able to implement it, or you can just assume that your target implementation understands it (as you would now).

It sounds more complicated because there's a new concept, but vocabularies are more modular and likely to remain a relatively small stable set than meta-schemas, which have never functioned well as a keyword extension system.

Reply:
To be clear, I am not suggesting that you add support for anything like this now!